iina/PlayerWindowController.swift

```diff
@@ -0,0 +1,516 @@
+  @IBOutlet weak var playSlider: NSSlider!
+  @IBOutlet weak var rightLabel: DurationDisplayTextField!
+  @IBOutlet weak var leftLabel: NSTextField!
+
+  /** Differentiate between single clicks and double clicks. */
+  internal var singleClickTimer: Timer?
+  internal var mouseExitEnterCount = 0
+
+  // Scroll direction
+
+  /** The direction of current scrolling event. */
+  enum ScrollDirection {
+    case horizontal
+    case vertical
+  }
+
+  internal var scrollDirection: ScrollDirection?
+
+  /** We need to pause the video when a user starts seeking by scrolling.
+   This property records whether the video is paused initially so we can
+   recover the status when scrolling finished. */
+  internal var wasPlayingWhenSeekBegan: Bool?
+  
+  /** Subclasses should set these value to true if the mouse is in some
+   special views (e.g. volume slider, play slider) before calling
+   `super.scrollWheel()` and set them back to false after calling
+   `super.scrollWheel()`.*/
+  internal var seekOverride = false
+  internal var volumeOverride = false
+
+  override func windowDidLoad() {
+    super.windowDidLoad()
+    loaded = true
+    
+    guard let window = window else { return }
+    
+    // Insert `menuActionHandler` into the responder chain
+    menuActionHandler = MainMenuActionHandler(playerCore: player)
+    let responder = window.nextResponder
+    window.nextResponder = menuActionHandler
+    menuActionHandler.nextResponder = responder
+    
+    window.initialFirstResponder = nil
+    window.titlebarAppearsTransparent = true
+    
+    setMaterial(Preference.enum(for: .themeMaterial))
+    
+    notificationCenter(.default, addObserverForName: .iinaMediaTitleChanged, object: player) { [unowned self] _ in
+        self.updateTitle()
+    }
+    
+    rightLabel.mode = Preference.bool(for: .showRemainingTime) ? .remaining : .duration
+    
+    updateVolume()
+    
+    observedPrefKeys.forEach { key in
+      UserDefaults.standard.addObserver(self, forKeyPath: key.rawValue, options: .new, context: nil)
+    }
+    
+    notificationCenter(.default, addObserverForName: .iinaFileLoaded, object: player) { [unowned self] _ in
+      self.updateTitle()
+    }
+    
+    NSWorkspace.shared.notificationCenter.addObserver(forName: NSWorkspace.willSleepNotification, object: nil, queue: nil, using: { [unowned self] _ in
+      if Preference.bool(for: .pauseWhenGoesToSleep) {
+        self.player.pause()
+      }
+    })
+  }
+  
+  deinit {
+    ObjcUtils.silenced {
+      for key in self.observedPrefKeys {
+        UserDefaults.standard.removeObserver(self, forKeyPath: key.rawValue)
+      }
+      for (center, observers) in self.notificationObservers {
+        for observer in observers {
+          center.removeObserver(observer)
+        }
+      }
+    }
+  }
+  
+  internal func notificationCenter(_ center: NotificationCenter, addObserverForName name: Notification.Name, object: Any? = nil, using block: @escaping (Notification) -> Void) {
```

Comment:
After some thought, I think this name evokes a delegate method. perhaps `addObserver(to notificationCenter: NotificationCenter, forName name: Notification.Name, object: Any?, using block: @escaping (Notification) -> Void)`?